File: backend/src/main/java/com/bangchat/data/entity/CharacterDetail.java
package com.bangchat.data.entity;

import lombok.Data;

/**
 * 角色行为细节实体，映射 character_detail 表。
 * 描述角色的行为模式和习惯，用于丰富 prompt 中的角色刻画。
 */
@Data
public class CharacterDetail {
    private Long id;
    private String characterId;
    private String detail;
    private Integer sortOrder;
}

File: bangchat/lib/models/character_detail.dart
/// 角色行为细节实体，映射 character_detail 表。
class CharacterDetail {
  final int? id;
  final String characterId;
  final String detail;
  final int sortOrder;

  const CharacterDetail({
    this.id,
    required this.characterId,
    required this.detail,
    this.sortOrder = 0,
  });

  factory CharacterDetail.fromMap(Map<String, dynamic> map) {
    return CharacterDetail(
      id: map['id'] as int?,
      characterId: (map['character_id'] as String?) ?? '',
      detail: (map['detail'] as String?) ?? '',
      sortOrder: (map['sort_order'] as int?) ?? 0,
    );
  }
}
